src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -13,10 +13,24 @@ namespace System.Net.NetworkInformation
     public class NetworkChange
     {
         private static NetworkAddressChangedEventHandler s_addressChangedSubscribers;
+        private static NetworkAvailabilityChangedEventHandler s_availabilityChangedSubscribers;
         private static volatile int s_socket = 0;
         private static readonly object s_lockObj = new object();
+        private static readonly object s_availabilityLock = new object();
```

Comment:
When is s_availabilityLock used vs when is s_lockObj used?  If both are needed, a comment on each would be helpful to indicate what it protects, and renaming s_lockObj to better indicate its purpose would also be helpful.